src/types.ts

```diff
@@ -133,23 +136,69 @@ type ProviderMethod = (
 ) => void;
 
 export type TransactionRequest = {
-    to?: unknown,
-    from?: unknown,
-    nonce?: unknown,
+  to?: unknown;
+  from?: unknown;
+  nonce?: unknown;
+
+  gasLimit?: unknown;
+  gasPrice?: unknown;
 
-    gasLimit?: unknown,
-    gasPrice?: unknown,
+  data?: unknown;
+  value?: unknown;
+  chainId?: unknown;
+};
+
+export interface EventData {
+  address: string;
+  blockHash: string;
+  blockNumber: string;
+  data: string;
+  logIndex: string;
+  removed: boolean;
+  topics: string[];
+  transactionHash: string;
+  transactionIndex: string;
+}
 
-    data?: unknown,
-    value?: unknown,
-    chainId?: unknown,
+export interface EventFilter {
+  address?: string;
+  topics?: Array<string>;
+  fromBlock?: string;
+  toBlock?: string;
+}
+
+export type RpcProviderTestCase = {
+  request: RpcProviderRequestBody;
+  response: string | RpcProviderLogEntry[];
+}[];
+
+export interface RpcProviderRequestBody {
+  data?: string;
+  to?: string;
+  fromBlock?: string;
+  toBlock?: string;
+  address?: string;
+  topics?: string[];
```

Comment:
All types you added here seem to be internal. Add `/** @internal */` tag to them.